### PR TITLE
Mobile optimize

### LINKED ISSUE #34: bug > checkout sidebar > user can checkout w/o items in cart
**Describe the bug**
A user can advance to their impact receipt even if they have no items in their cart

**To Reproduce**
1. Click on the shopping cart icon in nav without adding items
2. Click checkout
3. See empty impact receipt

**Expected behavior**
User gets an error message if they attempt to checkout when they have no items in their shopping cart, Error message should look like this:
[image]

**Screenshots**

**Devices & Browsers**
Behavior is consistent across all devices and browsers.

### PR DESCRIPTION
**Pull request by willyd332**
Adjusted all of the styling on landing page, item modal, and cart modal to work (i.e. not break and still be useable) with all sorts of screen sizes.

### CODE REVIEW

File: src/components/item-view/data-section/DataBreakdown.js [hunk 1]
```diff
@@ -15,48 +21,92 @@ const DataBreakdown = ({
...
 
   return (
-    <Grid container spacing={1} className={styles.breakdownBox}>
+    <Grid
+      container
+      spacing={2}
+      className={styles.breakdownBox}
+    >
       {body}
     </Grid>
```
Comment (nellbhealy) [inline src/components/item-view/data-section/DataBreakdown.js:114]: This gets cut off on dimensions for my laptop screen 

image

File: src/pages/receipt.js [hunk 2]
```diff
@@ -7,9 +7,21 @@ import ImpactAlternatives from '../components/receipt-view/receipt-section/Impac
...
 import SEO from '../components/seo';
 
-const ReceiptPage = ({ cartState }) => (
+const ReceiptPage = ({ cartState }) => {
+  const checkItems = Object.keys(cartState).map((itemKey) => {
+    if (cartState[itemKey] > 0) {
+      return true;
+    }
+    return null;
+  }).filter((item) => item);
```
Comment (nellbhealy) [inline src/pages/receipt.js:16]: I believe you could also achieve this with 
`const cartHasItems = !Object.keys(cartState).every((item) => !cartState[item])`
`if(!cartHasItems && typeof window !== 'undefined')`

File: src/components/item-view/data-section/DataBreakdown.js [hunk 3]
```diff
@@ -15,48 +21,92 @@ const DataBreakdown = ({
...
+            </Grid>
+            <Grid
+              item
+              xs={5}
+            >
+              <Typography
+                variant="h5"
+                className={styles.rawValue}
+              >
+                {text[type].rawValue}
```
Comment (nellbhealy) [inline src/components/item-view/data-section/DataBreakdown.js:88]: Messed up with Eutrophication 

image

**Review by nellbhealy — COMMENTED** [review]
Most of this works well 👍  Cart sidebar looks great on mobile.

Couple of issues with the detailed item modal

File: src/components/item-view/data-section/DataBreakdown.js (hunk 1 above)
Comment (willyd332) [reply src/components/item-view/data-section/DataBreakdown.js:114]: Got it, that's fixed now, you'll just have to scroll through the modal on narrow screens

File: src/components/item-view/data-section/DataBreakdown.js (hunk 3 above)
Comment (willyd332) [reply src/components/item-view/data-section/DataBreakdown.js:88]: Ok

**Pull request merged**